pkg/middleware/auth.go

```diff
@@ -146,6 +154,12 @@ func RoleAppPluginAuth(accessControl ac.AccessControl, ps pluginstore.Store, log
 			return
 		}
 
+		if !PageIsFeatureToggleEnabled(c.Req.Context(), c.Req.URL.Path) {
```

Comment:
It might be good to normalize the path to ensure `/a/my-plugin/settings` and `/a/my-plugin/settings/` resolve the same way